### PR TITLE
Fix 3 failures in doctest (4th is in dimod)

### PR DESCRIPTION
**Pull request by JoelPasvolsky**

### CODE REVIEW

File: hybrid/utils.py [hunk 1]
```diff
@@ -64,10 +64,10 @@ def bqm_reduced_to(bqm, variables, sample, keep_offset=True):
         This example reduces a 3-variable BQM to two variables.
 
         >>> import dimod           # Create a binary quadratic model
-        >>> bqm = dimod.BinaryQuadraticModel({}, {('a', 'b'): -1, ('b', 'c'): -1, ('c', 'a'): -1}, 0, 'BINARY')
+        >>> bqm = dimod.BinaryQuadraticModel({}, {('a', 'b'): -1.0, ('b', 'c'): -1.0, ('c', 'a'): -1.0}, 0.0, 'BINARY')
```
Comment (randomir) [inline hybrid/utils.py]: Why is this necessary for doctests to work?

File: hybrid/utils.py [hunk 2]
```diff
@@ -64,10 +64,10 @@ def bqm_reduced_to(bqm, variables, sample, keep_offset=True):
...
         >>> sample = {'a': 1, 'b': 1, 'c': 0}
         >>> bqm_reduced_to(bqm, ['a', 'b'], sample)
-        BinaryQuadraticModel({'a': 0.0, 'b': 0.0}, {('a', 'b'): -1}, 0.0, Vartype.BINARY)
+        BinaryQuadraticModel({'a': 0.0, 'b': 0.0}, {('a', 'b'): -1.0}, 0.0, Vartype.BINARY)
```
Comment (randomir) [inline hybrid/utils.py]: Are you sure variables in the first dictionary are going to be printed in the sorted order?

Comment (JoelPasvolsky) [reply hybrid/utils.py]: That's a good point, it seems to me they always are but it is a dict

Comment (arcondello) [reply hybrid/utils.py]: BQMs are unordered, other than to the extend all dics are ordered in py3.6+ (RTD currently uses 3.7)

File: hybrid/utils.py (hunk 1 above)
Comment (JoelPasvolsky) [reply hybrid/utils.py]: doctest was failing because when the biases are set as int that's preserved so it was barfing on the float results

File: hybrid/utils.py (hunk 2 above)
Comment (randomir) [reply hybrid/utils.py]: Exactly. So either we use OrderedBQMs implicitly (not transparent), or we change all doctests (nor nice nor practical), or we accept tests will fail for users running them locally on older Pythons.

I think we can live with that last option.

File: hybrid/utils.py (hunk 1 above)
Comment (randomir) [reply hybrid/utils.py]: I don't understand what do you mean by that.

In this case for integer couplings, couplings in the reduced bqm are also integers.

In other words, there's no need to upcast the input, you can just use integer output:
```
BinaryQuadraticModel({'b': 0, 'a': 0}, {('b', 'a'): -1}, 0, Vartype.BINARY)
```

Comment (randomir) [reply hybrid/utils.py]: I actually answered my initial question with that last comment. To match (mixed) float output that was already present in the docstring, you upcasted everything to float.

That's fine. Although unnecessary. :smile:
